trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/plugin/config/PluginIdentifierConfig.java

```diff
@@ -11,16 +11,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.trino.aws.proxy.spi;
+package io.trino.aws.proxy.spi.plugin.config;
 
-import com.google.inject.Module;
-
-public interface TrinoAwsProxyServerPlugin
+public interface PluginIdentifierConfig<T>
```

Comment:
This will change to always just return `String` right? i.e. it doesn't need a type parameter.

Reply:
I wasn't sure if we want to do this or not, its equivalent to just using strings as the plugin keys. Do you see a middle ground?

Reply:
IMO - the extra objects, etc. add a lot of weight with not much value. I'd prefer just strings. I don't think there's a better way if we're going to use a Config-based mechanism.

Reply:
Updated, using strings now.

(Will squash all the commits before merging, I like having a history when working on a pr)